### PR TITLE
Enables the GD support for resolution options group for PHP 7 >= 7.2, PHP 8.0.

### PR DESCRIPTION
**Pull request by gutocf**
Example:

```
<?php

use Imagine\Image\ImageInterface;

$imagine = new Imagine\Gd\Imagine();

$options = array(
    'resolution-units' => ImageInterface::RESOLUTION_PIXELSPERINCH,
    'resolution-x' => 300,
    'resolution-y' => 300,
);

$imagine->open('/path/to/image.jpg')->save('/path/to/image.jpg', $options);
```

### CODE REVIEW

File: src/Gd/Image.php [hunk 1]
```diff
@@ -727,6 +727,18 @@ private function saveOrOutput(Format $format, array $options, $filename = null)
...
+        if (isset($options['resolution-units']) && isset($options['resolution-x']) && function_exists('imageresolution')) {
+            $resolution_x = $options['resolution-x'];
+            $resolution_y = isset($options['resolution-y']) ? $options['resolution-y'] : $resolution_x;
+
+            if ($options['resolution-units'] === ImageInterface::RESOLUTION_PIXELSPERCENTIMETER) {
+                $resolution_x = $resolution_x * ImageInterface::RESOLUTION_PPC_TO_PPI_MULTIPLIER;
+                $resolution_y = $resolution_y * ImageInterface::RESOLUTION_PPC_TO_PPI_MULTIPLIER;
+            }
+
+            imageresolution($this->resource, $resolution_x, $resolution_y);
```
Comment (ausi) [inline src/Gd/Image.php:739]:
```suggestion
            $resolutionX = $options['resolution-x'];
            $resolutionY = isset($options['resolution-y']) ? $options['resolution-y'] : $resolutionX;

            if ($options['resolution-units'] === ImageInterface::RESOLUTION_PIXELSPERCENTIMETER) {
                $resolutionX *= ImageInterface::RESOLUTION_PPC_TO_PPI_MULTIPLIER;
                $resolutionY *= ImageInterface::RESOLUTION_PPC_TO_PPI_MULTIPLIER;
            }

            imageresolution($this->resource, $resolutionX, $resolutionY);
```

I think the coding style of Imagine is to use camel case variable names (but I’m not sure ☺️)

**Review by ausi — APPROVED** [review]
Apart from the coding style, this looks great! ❤️

File: src/Gd/Image.php (hunk 1 above)
Comment (gutocf) [reply src/Gd/Image.php]: You're right. Thanks for your review.

**Review by ausi — APPROVED** [review]

**Comment by bartthys** [comment]
Hi @gutocf @ausi I'm looking for exactly this feature, I see the PR was approved but it was never merged?

**Comment by ausi** [comment]
@bartthys I’m not a maintainer of this repository, so my approval has no “official” meaning ☺️

**Comment by ausi** [comment]
In the meantime you can workaround that by using something like this in your code:

```php
if ($yourImage instanceof \Imagine\Gd\Image) {
	imageresolution($yourImage->getGdResource(), $resolutionXppi, $resolutionYppi);
}
```

**Comment by mlocati** [comment]
Superseded by #866

**Pull request closed without merging**